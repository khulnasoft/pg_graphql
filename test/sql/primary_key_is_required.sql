begin;

    savepoint a;

    create table account(
        id serial primary key
    );

    -- Should be visible because it has a primary ky
    select jsonb_pretty(
        graphql.resolve($$
        {
          __type(name: "Account") {
            name
          }
        }
        $$)
    );

    rollback to savepoint a;

    create table account(
        id serial
    );

    -- Should NOT be visible because it does not have a primary ky
    select jsonb_pretty(
        graphql.resolve($$
        {
          __type(name: "Account") {
            name
          }
        }
        $$)
    );

rollback;
