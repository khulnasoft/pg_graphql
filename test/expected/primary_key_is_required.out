begin;
    savepoint a;
    create table account(
        id serial primary key
    );
    -- Should be visible because it has a primary ky
    select jsonb_pretty(
        graphql.resolve($$
        {
          __type(name: "Account") {
            name
          }
        }
        $$)
    );
         jsonb_pretty          
-------------------------------
 {                            +
     "data": {                +
         "__type": {          +
             "name": "Account"+
         }                    +
     }                        +
 }
(1 row)

    rollback to savepoint a;
    create table account(
        id serial
    );
    -- Should NOT be visible because it does not have a primary ky
    select jsonb_pretty(
        graphql.resolve($$
        {
          __type(name: "Account") {
            name
          }
        }
        $$)
    );
      jsonb_pretty      
------------------------
 {                     +
     "data": {         +
         "__type": null+
     }                 +
 }
(1 row)

rollback;
